Jenkinsfile

```diff
@@ -490,6 +506,107 @@ az aks create                      \
+                                    def adminEmail = "${clusterName}@${parentZone}"
+                                    def dnsZone = "${dnsPrefix}.${parentZone}"
+
+                                    try {
+                                        runIntegrationTest(platform, "eks --user-pool-id=eu-central-1_P7sIQ92Cd --config=${clusterName}-autogen.json --dns-zone=${dnsZone} --email=${adminEmail}", "--dns-suffix ${dnsZone}") {
+                                            // Create EKS cluster: requires AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment
+                                            // variables to reference a user in AWS with the correct privileges to create an EKS
+                                            // cluster: https://github.com/weaveworks/eksctl/issues/204#issuecomment-450280786
+                                            container('eksctl') {
+                                                sh "eksctl create cluster --name ${clusterName} --version ${kversion} --node-type m5.large --nodes 3 --nodes-min 3 --nodes-max 3 --timeout 60m --kubeconfig \$KUBECONFIG --tags 'platform=${platform},branch=${BRANCH_NAME},build=${BUILD_URL}'"
+                                                withEnv(["PATH+KTOOL=${tool 'kubectl'}"]) {
+                                                    sh "kubectl create clusterrolebinding cluster-system-anonymous --clusterrole cluster-admin --user system:anonymous"
+                                                }
+                                                waitForRollout("kube-system", 30)
+                                            }
+
+                                            // NB: writeJSON doesn't work without approvals(?)
+                                            // See https://issues.jenkins-ci.org/browse/JENKINS-44587
+                                            writeFile([file: 'kubeprod-manifest.jsonnet', text: """
+(import "manifests/platforms/eks.jsonnet") {
+  config:: import "${clusterName}-autogen.json",
+  letsencrypt_environment: "staging",
+  prometheus+: import "tests/testdata/prometheus-crashloop-alerts.jsonnet",
+}
+"""
+                                            ])
+                                        }{
+                                            // update glue records in parent zone
+                                            def nameServers = []
+                                            container('aws') {
+                                                withEnv(["PATH+JQ=${tool 'jq'}"]) {
+                                                    def output = sh(returnStdout: true, script: """
+id=\$(aws route53 list-hosted-zones-by-name --dns-name "${dnsZone}" --max-items 1 | jq -r .HostedZones[0].Id)
+aws route53 get-hosted-zone --id "\$id" --query DelegationSet.NameServers
+"""
+                                                    )
+                                                    for (ns in readJSON(text: output)) {
```

Comment:
Why is this a loop?  I think we can just do `nameServers = readJSON(text: output)`

Reply:
Oops, sorry.  Looks like `readJSON` returns a `JSONArray`, but the `insertGlueRecords` function requires an `ArrayList`.  Need to put the loop back (boo!), or change `insertGlueRecords` to take a `java.util.List` (yay!) instead.

Reply:
Fixed.